src/foraging_gui/Foraging.py

```diff
@@ -109,6 +110,21 @@ def __init__(self, parent=None,box_number=1,start_bonsai_ide=True):
         # Load User interface
         self._LoadUI()
 
+        # create AINDBehaviorSession model to be used and referenced for session info
+        self.behavior_session_model = AindBehaviorSessionModel(
+            experiment=self.Task.currentText(),
+            experimenter=[self.Experimenter.text()],
+            date=datetime.now(),   # update when folders are created
+            root_path='',         # update when created
+            session_name= '',   # update when date and subject are filled in
+            subject=self.ID.text(),
+            experiment_version=foraging_gui.__version__,
+            notes=self.ShowNotes.toPlainText(),
+            commit_hash= subprocess.check_output(['git', 'rev-parse', 'HEAD']).decode('ascii').strip(),
+            allow_dirty_repo=subprocess.check_output(['git','diff-index','--name-only', 'HEAD']).decode('ascii').strip() != '',
```

Comment:
I think we want "allow_dirty_repo" to be "True". Since we are allowing people to run with dirty code. 

The subprocess call you make will report if the repo is dirty, which is different from what the schema is asking https://allenneuraldynamics.github.io/Aind.Behavior.Services/api.base/session.html
@bruno-f-cruz please clarify if I'm wrong.

Reply:
Yep. `allow_dirty` is True, means you intend people to be able to run the code with uncommitted changes and potentially untracked files.

Reply:
That line should be a boolean since I'm evaluating if its not an empty string. I thought if the repo was dirty, then set it to True and if not, False. I grabbed the logic for that line from [here ](https://github.com/AllenNeuralDynamics/dynamic-foraging-task/blob/60f8beb5cf259b096e71331217b2a631d1eaeb43/src/foraging_gui/Foraging.py#L4715C5-L4715C40)

Reply:
Oh sorry @micahwoodard I missed that you were checking for equality with the empty string. 

So your code will set `allow_dirty` to False if the repo is clean. And `allow_dirty` to True if the repo is dirty. Thats fine, given that we are allowing people to run dirty.